src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -889,6 +889,19 @@ private NamedTypeSymbol ConstructNamedTypeUnlessTypeArgumentOmitted(CSharpSyntax
 
             Debug.Assert(members.Count > 0);
 
+            if (!hasErrors && members[0] is LocalFunctionSymbol)
+            {
+                Debug.Assert(members.Count == 1 && members[0].Locations.Length == 1);
```

Comment:
It looks to me like `LookupResult.MergeEqual` can return a list of multiple symbols.